Makefile

```diff
@@ -1,38 +1,154 @@
+	vulndb
+
+DOCS = \
+	CODE_OF_CONDUCT.md \
+	CONTRIBUTING.md \
+	HOWTO.md \
+	LICENSE \
+	README.md
+
+GO = go
+GOOS = $(shell $(GO) env GOOS)
+GOARCH = $(shell $(GO) env GOARCH)
+
+TAR = tar
+ZIP = zip
+INSTALL = install
+
+# Compile all tools.
+all: $(TOOLS)
+
+# Compile TOOLS to ./build/bin/$tool using GOOS and GOARCH.
+$(TOOLS):
+	GOOS=$(GOOS) GOARCH=$(GOARCH) $(GO) build $(GOFLAGS) -o ./build/bin/$@ ./cmd/$@
+
+# Check/fetch all dependencies.
+deps:
+	GOOS=$(GOOS) GOARCH=$(GOARCH) $(GO) get -v -d ./...
+
+# install installs tools and documentation.
+# The install target is used by rpm and deb builders.
 install:
-	install -d $(DESTDIR)/usr/bin
-	install -p -m 0755 ./cpe2cve $(DESTDIR)/usr/bin/cpe2cve
-	install -p -m 0755 ./csv2cpe $(DESTDIR)/usr/bin/csv2cpe
-	install -p -m 0755 ./nvdsync $(DESTDIR)/usr/bin/nvdsync
-	install -p -m 0755 ./rpm2cpe $(DESTDIR)/usr/bin/rpm2cpe	 
-
-archive:
-	touch $(TGZ)
-	tar czf $(TGZ) --exclude=$(TGZ) --transform s/$(NAME)/$(PKG)/ ../$(NAME)
+	# tools
+	$(INSTALL) -d $(DESTDIR)/usr/bin
+	for tool in $(TOOLS); do $(INSTALL) -p -m 0755 ./build/bin/$$tool $(DESTDIR)/usr/bin/$$tool; done
+	# docs
+	$(INSTALL) -d $(DESTDIR)/usr/share/doc/nvdtools
+	for doc in $(DOCS); do $(INSTALL) -p -m 0644 $$doc $(DESTDIR)/usr/share/doc/nvdtools/$$doc; done
+
+DIST_NAME = $(NAME)-$(VERSION)
+DIST_DIR = build/$(DIST_NAME)
+
+# binary_dist creates a local binary distribution in DIST_DIR.
+binary_dist: $(TOOLS)
+	[ -d $(DIST_DIR) ] || mkdir -p $(DIST_DIR)/doc
+	cp $(DOCS) $(DIST_DIR)/doc
+	mv build/bin $(DIST_DIR)/bin
+
+# binary_tar creates tarball of binary distribution.
+binary_tar: binary_dist
+	[ -d build/tgz ] || mkdir build/tgz
+	cd build && $(TAR) czf tgz/$(NAME)_$(GOOS)_$(GOARCH)-$(VERSION).tar.gz $(DIST_NAME)
+	rm -rf $(DIST_DIR)
+
+# binary_zip creates zip of binary distribution.
+binary_zip: binary_dist
+	[ -d build/zip ] || mkdir build/zip
+	cd build && $(ZIP) -r zip/$(NAME)_$(GOOS)_$(GOARCH)-$(VERSION).zip $(DIST_NAME)
+	rm -rf $(DIST_DIR)
+
+# binary_deb creates debian package.
+#
+# Requires GOPATH and dependencies available to compile nvdtools.
+# Must set version to build: make binary_deb VERSION=1.0
+binary_deb:
+	VERSION=$(VERSION) dpkg-buildpackage -rfakeroot -uc -us
+	[ -d build/deb ] || mkdir -p build/deb
+	mv ../$(NAME)*.deb build/deb/
+
+# archive_tar creates tarball of the source code.
+archive_tar:
+	[ -d build/tgz ] || mkdir -p build/tgz
+	$(TAR) czf build/tgz/$(DIST_NAME).tar.gz \
+		--exclude=build \
+		--exclude=release \
+		--exclude=.git \
+		--exclude=.travis.yml \
+		--transform s/./$(DIST_NAME)/ \
+		.
+
+# binary_rpm creates rpm package.
+#
+# Requires GOPATH and dependencies available to compile nvdtools.
+# Must set version to build: make binary_rpm VERSION=1.0
+binary_rpm: archive_tar
+	[ -d build/rpm/SOURCES ] || mkdir -p build/rpm/SOURCES
+	mv build/tgz/$(DIST_NAME).tar.gz build/rpm/SOURCES/
+	rpmbuild -ba \
+		--define="_topdir $(PWD)/build/rpm" \
+		--define="_version $(VERSION)" \
```

Comment:
We probably need to start maintaining a changelog. The we could grep the version from it... Just a thought, not related to this commit specifically.

Reply:
Both rpm and debian packages need changelog. We should probably auto-generate them from the commit log or something along those lines.

The version needs more love: we can now easily build packages with any version by passing VERSION to make. This is how we use git tags in .travis.yml for automated releases.

What is missing is having a -version in all binaries.